avocado/core/job.py

```diff
@@ -548,7 +548,7 @@ def create_test_suite(self):
         refs = self.config.get('run.references')
         if not refs:
             refs = self.config.get('nrun.references')
-        runner_name = self.config.get('test_runner', 'runner')
+        runner_name = self.config.get('run.test_runner', 'runner')
```

Comment:
IMO, one of the advantages of having a `settings.register_option()` is to avoid spreading around the code the "default values" since that the default value is registered during this method call.

I understand that this will also be consumed by the Job API and in some cases, the `register_option()` will not be called. 

All other calls using the new settings modules are assuming that the value is already there. I had understood this behavior in one of our discussions about this. I think that there are only one or two exceptions to that.

I'm not sure, but I have a feeling that a complete and explicit config file is better when passing to the Job API instead of a minimal config with implicit values.

As I said it is just a feeling and my only concern here is having back all the default values spread around the source code. 

What do you think?